mypy/nodes.py

```diff
@@ -334,7 +336,16 @@ def __init__(self, arguments: List[Argument], body: 'Block',
         arg_kinds = [arg.kind for arg in self.arguments]
         self.max_pos = arg_kinds.count(ARG_POS) + arg_kinds.count(ARG_OPT)
         self.body = body
-        self.type = typ
+        if typ is not None:
+            self.type = typ
+        elif Options.get_options().implicit_any:
```

Comment:
I have some style quibbles about this. First, I wouldn't make `mypy.nodes` know about command line options. I'd move this logic to the parser, since the parser already uses a bunch of options. The `nodes` module is intended to be a rather stupid data structure. Second, I'd rather not use global state for options, and I wouldn't want to make the parser know about the details of the command line interface in the interest of modularity. If we'd want to ship the parser as a separate package, it would have to include information about all mypy command line options with it, which would be a little ugly. The straightforward thing to do is to pass the flag as an extra argument to `mypy.build.build`, though then you have to propagate it to the parser. This has the benefit of making it explicit which parts of the implementation depend on which option.